coderd/provisionerdserver/provisionerdserver_test.go

```diff
@@ -1804,6 +1802,64 @@ func TestNotifications(t *testing.T) {
 			})
 		}
 	})
+
+	t.Run("Manual build failed, template admins notified", func(t *testing.T) {
+		t.Parallel()
+
+		ctx := context.Background()
+
+		// given
+		notifEnq := &testutil.FakeNotificationsEnqueuer{}
+		srv, db, ps, pd := setup(t, true /* ignoreLogErrors */, &overrides{notificationEnqueuer: notifEnq})
+
+		templateAdmin := dbgen.User(t, db, database.User{RBACRoles: []string{codersdk.RoleTemplateAdmin}})
+		user := dbgen.User(t, db, database.User{})
+
+		template := dbgen.Template(t, db, database.Template{
+			Name: "template", Provisioner: database.ProvisionerTypeEcho, OrganizationID: pd.OrganizationID,
+		})
+		workspace := dbgen.Workspace(t, db, database.Workspace{
+			TemplateID: template.ID, OwnerID: user.ID, OrganizationID: pd.OrganizationID,
+		})
+		version := dbgen.TemplateVersion(t, db, database.TemplateVersion{
+			OrganizationID: pd.OrganizationID, TemplateID: uuid.NullUUID{UUID: template.ID, Valid: true}, JobID: uuid.New(),
+		})
+		build := dbgen.WorkspaceBuild(t, db, database.WorkspaceBuild{
+			WorkspaceID: workspace.ID, TemplateVersionID: version.ID, InitiatorID: user.ID, Transition: database.WorkspaceTransitionDelete, Reason: database.BuildReasonInitiator,
+		})
+		job := dbgen.ProvisionerJob(t, db, ps, database.ProvisionerJob{
+			FileID:         dbgen.File(t, db, database.File{CreatedBy: user.ID}).ID,
+			Type:           database.ProvisionerJobTypeWorkspaceBuild,
+			Input:          must(json.Marshal(provisionerdserver.WorkspaceProvisionJob{WorkspaceBuildID: build.ID})),
+			OrganizationID: pd.OrganizationID,
+		})
+		_, err := db.AcquireProvisionerJob(ctx, database.AcquireProvisionerJobParams{
+			OrganizationID: pd.OrganizationID,
+			WorkerID:       uuid.NullUUID{UUID: pd.ID, Valid: true},
+			Types:          []database.ProvisionerType{database.ProvisionerTypeEcho},
+		})
+		require.NoError(t, err)
+
+		// when
+		_, err = srv.FailJob(ctx, &proto.FailedJob{
+			JobId: job.ID.String(), Type: &proto.FailedJob_WorkspaceBuild_{WorkspaceBuild: &proto.FailedJob_WorkspaceBuild{State: []byte{}}},
+		})
+		require.NoError(t, err)
+
+		// then
+		require.Len(t, notifEnq.Sent, 1)
+		assert.Equal(t, notifEnq.Sent[0].UserID, templateAdmin.ID)
+		assert.Equal(t, notifEnq.Sent[0].TemplateID, notifications.TemplateWorkspaceManualBuildFailed)
+		assert.Contains(t, notifEnq.Sent[0].Targets, template.ID)
+		assert.Contains(t, notifEnq.Sent[0].Targets, workspace.ID)
+		assert.Contains(t, notifEnq.Sent[0].Targets, workspace.OrganizationID)
+		assert.Contains(t, notifEnq.Sent[0].Targets, user.ID)
+		assert.Equal(t, workspace.Name, notifEnq.Sent[0].Labels["name"])
+		assert.Equal(t, template.Name, notifEnq.Sent[0].Labels["template_name"])
+		assert.Equal(t, version.Name, notifEnq.Sent[0].Labels["template_version_name"])
+		assert.Equal(t, user.Username, notifEnq.Sent[0].Labels["initiator"])
+		assert.Equal(t, user.Username, notifEnq.Sent[0].Labels["workspace_owner_username"])
```

Comment:
I feel like these assertions could be passed into the previous table-based test "Workspace build failed"?

Reply:
There is a significant difference as here we expect notifications sent only to template admins, we don't care about workspace owners. I tried to squeeze this unit test as much as possible.